.github/workflows/software-environments-nightly.yml

```
@@ -0,0 +1,76 @@
+name: Create software environments
+
+on:
+  schedule:
+    # Runs at 01:00 every day after pre-release
+    - cron: "0 1 * * *"
+
+jobs:
+  build:
+    name: Python ${{ matrix.python-version }}, nightly build, ${{ matrix.server }}
+    runs-on: ubuntu-latest
+    strategy:
+      fail-fast: false
+      matrix:
+        python-version: ["3.8", "3.9", "3.10"]
+        server: ["https://staging.coiledhq.com"]
+        include: #leaving this logic as we build nightly soft env for prod. 
+          - server: "https://staging.coiledhq.com"
+            token-name: COILED_STAGING_BENCHMARK_BOT_TOKEN 
+
+    steps:
+      - name: Checkout source
+        uses: actions/checkout@v2
+
+      - name: Set up Python
+        uses: actions/setup-python@v1
+
+      - name: Install Coiled
+        run: python -m pip install coiled
+
+      - name: Create Runtime software environment environment variables
+        run: |
+          # conda search for the coiled-runtime latest nightly
+          arr_nightly=($(conda search --override-channels -c coiled/label/dev coiled-runtime | tail -n 1))
+          
+          # extract coiled-runtime nightly version / build
+          export RUNTIME_NIGHTLY_VERSION=${arr_nightly[1]}
+          export RUNTIME_NIGHTLY_VERSION_FORMATTED=$(echo $RUNTIME_NIGHTLY_VERSION | sed 's/\./-/g' )
+          echo "Latest coiled-runtime nightly version formatted $RUNTIME_NIGHTLY_VERSION_FORMATTED"
+
+          # Coiled software environment names can't contain "." or uppercase characters
+          export PYTHON_VERSION_FORMATTED=$(echo "${{ matrix.python-version }}" | sed 's/\.//g' )
+          export COILED_SOFTWARE_ENV_NAME=coiled-runtime-$RUNTIME_NIGHTLY_VERSION_FORMATTED-py$PYTHON_VERSION_FORMATTED
+          export ENV_FILENAME=coiled-runtime-nightly-py$PYTHON_VERSION_FORMATTED
+
+          # Put variables into $GITHUB_ENV so it can be used in subsequent workflow steps
+          echo COILED_SOFTWARE_ENV_NAME=$COILED_SOFTWARE_ENV_NAME >> $GITHUB_ENV
+          echo ENV_FILENAME=$ENV_FILENAME >> $GITHUB_ENV
+          echo RUNTIME_NIGHTLY_VERSION=$RUNTIME_NIGHTLY_VERSION >> $GITHUB_ENV 
+      
+      - name: Find and Replace nightly
```

Nitpick: This name seems a bit more clear about the intent of this step 

```suggestion
      - name: Pin latest nightly coiled-runtime
```